### PR TITLE
TST: Add simple BNB regression tests

### PR DESCRIPTION
**Pull request by BenjaminBossan**
These are very basic and simplistic regression tests for bnb. Their
purpose is to ensure that there is no unnoticed change in bnb that leads
to different outputs. There is no check for "correctness", just that the
results haven't changed.

Eventually, this workflow should be improved and moved to the bnb repo.

Before merging, it would be good if someone can double check that these
tests pass for them as well (GPU CI does not run for PRs).

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks for adding the regression tests!

**Comment by BenjaminBossan** [comment]
*not stale*

File: tests/bnb/test_bnb_regression.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    bnb_config = BitsAndBytesConfig(
+        load_in_4bit=True,
+        bnb_4bit_use_double_quant=False,
+        bnb_4bit_compute_dtype=torch.float16,
+    )
+    model = AutoModelForCausalLM.from_pretrained(
+        "facebook/opt-350m",
+        quantization_config=bnb_config,
+        torch_dtype=torch.float32,
+    ).eval()
```
Comment (younesbelkada) [inline tests/bnb/test_bnb_regression.py:127]:
```suggestion
    model = AutoModelForCausalLM.from_pretrained(
        "facebook/opt-350m",
        quantization_config=bnb_config,
        torch_dtype=torch.float32,
    )
```

File: tests/bnb/test_bnb_regression.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="No CUDA device available.")
+def test_opt_350m_8bit():
+    torch.manual_seed(0)
+    bnb_config = BitsAndBytesConfig(load_in_8bit=True)
+    model = AutoModelForCausalLM.from_pretrained(
+        "facebook/opt-350m",
+        quantization_config=bnb_config,
+        torch_dtype=torch.float32,
+    ).eval()
```
Comment (younesbelkada) [inline tests/bnb/test_bnb_regression.py:81]:
```suggestion
    model = AutoModelForCausalLM.from_pretrained(
        "facebook/opt-350m",
        quantization_config=bnb_config,
        torch_dtype=torch.float32,
    )
```

File: tests/bnb/test_bnb_regression.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    bnb_config = BitsAndBytesConfig(
+        load_in_4bit=True,
+        bnb_4bit_use_double_quant=False,
+        bnb_4bit_compute_dtype=torch.float32,
+    )
+    model = AutoModelForCausalLM.from_pretrained(
+        "facebook/opt-350m",
+        quantization_config=bnb_config,
+        torch_dtype=torch.float32,
+    ).eval()
```
Comment (younesbelkada) [inline tests/bnb/test_bnb_regression.py:62]:
```suggestion
    model = AutoModelForCausalLM.from_pretrained(
        "facebook/opt-350m",
        quantization_config=bnb_config,
        torch_dtype=torch.float32,
    )
```

**Pull request merged**